native/src/main/scala/sqs4s/auth/Credentials.scala

```diff
@@ -59,23 +57,22 @@ object Credentials {
     * Instance profile credentials– used on EC2 instances, and delivered through
     * the Amazon EC2 metadata service.
     *
-    * @param client http client
-    * @param ttl session token's time to live
+    * @param client        http client
+    * @param ttl           session token's time to live
     * @param refreshBefore a small duration to refresh the token before it expires
     * @tparam F an effect which represents the side effects
     * @return
     */
-  def chainResource[F[_]: Concurrent: Clock: Timer](
+  def chain[F[_]: Concurrent: Clock: Timer](
     client: Client[F],
     ttl: FiniteDuration = 6.hours,
     refreshBefore: FiniteDuration = 5.minutes
-  ): Resource[F, Credentials[F]] = {
-    Resource {
-      envVar[F].map(cred => cred -> ().pure[F]).handleErrorWith(_ =>
-        sysProp[F].map(cred => cred -> ().pure[F])).handleErrorWith(_ =>
-        instanceMetadata[F](client, ttl, refreshBefore))
-    }
-  }
+  ): Resource[F, Credentials[F]] =
+    envVar[F] orElse sysProp[F] orElse instanceMetadata[F](
+      client,
+      ttl,
+      refreshBefore
+    )
```

Comment:
I forgot to mention before that if all fail, we would get the error message of the last one.
So, in this case, it would be better to do something like this:

```scala
(
  envVar[F] orElse sysProp[F] orElse instanceMetadata[F](
    client,
    ttl,
    refreshBefore
  )
).handleErrorWith(_ => Sync[F].raiseError[Credential](NoValidAuthMethodError))
```